src/main/haskell/language-kore/src/Data/Kore/ASTUtils/Substitution.hs

```diff
@@ -0,0 +1,129 @@
+{-|
+Module      : Data.Kore.ASTUtils.Substitution
+Description : Substitute phi_1 for phi_2, avoiding capture
+Copyright   : (c) Runtime Verification, 2018
+License     : UIUC/NCSA
+Maintainer  : phillip.harris@runtimeverification.com
+Stability   : experimental
+Portability : portable
+-}
+
+{-# LANGUAGE MultiParamTypeClasses  #-}
+{-# LANGUAGE FunctionalDependencies #-}
+{-# LANGUAGE Rank2Types             #-}
+{-# LANGUAGE AllowAmbiguousTypes    #-}
+{-# LANGUAGE FlexibleContexts       #-}
+{-# LANGUAGE BangPatterns           #-}
+{-# LANGUAGE LambdaCase             #-}
+{-# LANGUAGE DeriveFunctor          #-}
+{-# LANGUAGE DeriveFoldable         #-}
+{-# LANGUAGE DeriveTraversable      #-}
+{-# LANGUAGE TypeFamilies           #-}
+{-# LANGUAGE PatternSynonyms        #-}
+{-# LANGUAGE ScopedTypeVariables    #-}
+{-# LANGUAGE TypeApplications       #-}
+{-# OPTIONS_GHC -Wno-unused-matches #-}
+{-# OPTIONS_GHC -Wno-missing-signatures #-}
+{-# OPTIONS_GHC -Wno-name-shadowing #-}
+
+module Data.Kore.ASTUtils.Substitution where
+
+
+import           Control.Lens
+
+import qualified Data.Set as S
+import           Data.Fix
+import           Data.List
+
+import           Data.Kore.AST.Common
+import           Data.Kore.AST.MetaOrObject
+import           Data.Kore.AST.PureML
+
+import           Data.Kore.ASTUtils.SmartConstructors
+
+
+subst
```

Comment:
Old and new seems fine. Substituted and subtituting is also fine. Wrapping them in data types so that people calling your method do not mix the arguments is even better, but I can't usually convince others that it's worth doing. Anyway, the issue was that you had three patterns as arguments and it was not obvious which is which.